internal/action/infocomplete.go

```diff
@@ -81,9 +82,40 @@ func colorschemeComplete(input string) (string, []string) {
 func filetypeComplete(input string) (string, []string) {
 	var suggestions []string
 
-	for _, f := range config.ListRuntimeFiles(config.RTSyntax) {
-		if strings.HasPrefix(f.Name(), input) {
-			suggestions = append(suggestions, f.Name())
+	// Since we need to complete the file type instead of the file name
+	// we need to parse the RTSyntax* files and search for a matching FileType.
+	// The following logic can be simplified to the previos state again,
+	// in the moment the file names will represent the file types directly.
+	for _, f := range config.ListRealRuntimeFiles(config.RTSyntax) {
+		data, err := f.Data()
+		if err != nil {
+			continue
+		}
+		header, err := highlight.MakeHeaderYaml(data)
+		if err != nil {
+			continue
+		}
+		if strings.HasPrefix(header.FileType, input) {
+			suggestions = append(suggestions, header.FileType)
```

Comment:
If the user (wrongfully) adds a file with `filetype: "unknown"` or `filetype: "off"` to `~/.config/micro/syntax`, we get duplicated suggestions for `unknown` or `off`.